.JamesLau-fe-project/src/Components/ArticlePage.jsx

```diff
@@ -1,33 +1,47 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../api";
-import Comments from "./Comments"
+import { getArticleById, patchVotesByArticleId } from "../api";
+
+import Comments from "./Comments";
 const ArticlePage = () => {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const [loading, setLoading] = useState(true);
+  const [currentVotes, setCurrentVotes] = useState(0);
 
   useEffect(() => {
     getArticleById(article_id).then((data) => {
       setArticle(data);
+      setCurrentVotes(data.votes);
       setLoading(false);
     });
   }, [article_id]);
 
+  const handleVote = (voteToHandle) => {
+    setCurrentVotes((prevVotes) => prevVotes + voteToHandle);
+    patchVotesByArticleId(article_id, voteToHandle).then((newData) => {
```

Comment:
As you're changing the currentVotes to optimistically render, you don't need to setArticle to the returned data. As you don't have anything re rendering if article changes, this isn't actually doing anything. The api call has happened, yes and the votes persist, you just don't need the then block.

Reply:
However, it would be great to have a catch block on this api call as if I set my connection to offline I can click on the like button and it 'appears' to work even though no api call can happen. With a catch you can handle telling the user something is wrong.